src/test/optimizer/strategy/in_expression_rewrite_rule_test.cpp

```diff
@@ -0,0 +1,297 @@
+    duplicate_element_in_expression = in_(col_a, list_(1, 2, 1));
+    different_types_on_left_and_right_side_expression = in_(col_b, list_(1, 2));
+    different_types_on_right_side_expression = in_(col_a, list_(1, 2.0f));
+    null_in_expression = in_(col_a, list_(1, NULL_VALUE));
+
+    auto hundred_elements = std::vector<std::shared_ptr<AbstractExpression>>{};
+    for (auto i = 0; i < 100; ++i) hundred_elements.emplace_back(value_(i));
+    hundred_element_in_expression = std::make_shared<InExpression>(PredicateCondition::In, col_a,
+                                                                   std::make_shared<ListExpression>(hundred_elements));
+  }
+
+ public:
+  std::shared_ptr<MockNode> node;
+  std::shared_ptr<AbstractExpression> col_a, col_b, single_element_in_expression, five_element_in_expression,
+      five_element_not_in_expression, hundred_element_in_expression, duplicate_element_in_expression,
+      different_types_on_left_and_right_side_expression, different_types_on_right_side_expression, null_in_expression;
+};
+
+TEST_F(InExpressionRewriteRuleTest, ExpressionEvaluatorStrategy) {
+  // With the ExpressionEvaluator strategy chosen, no modifications should be made
+  auto rule = std::make_shared<InExpressionRewriteRule>();
+  rule->strategy = InExpressionRewriteRule::Strategy::ExpressionEvaluator;
+
+  {
+    const auto input_lqp = PredicateNode::make(single_element_in_expression, node);
+    const auto result_lqp = StrategyBaseTest::apply_rule(rule, input_lqp);
+    EXPECT_EQ(result_lqp, input_lqp);
+  }
+
+  {
+    const auto input_lqp = PredicateNode::make(five_element_in_expression, node);
+    const auto result_lqp = StrategyBaseTest::apply_rule(rule, input_lqp);
+    EXPECT_EQ(result_lqp, input_lqp);
+  }
+
+  {
+    const auto input_lqp = PredicateNode::make(five_element_not_in_expression, node);
+    const auto result_lqp = StrategyBaseTest::apply_rule(rule, input_lqp);
+    EXPECT_EQ(result_lqp, input_lqp);
+  }
+
+  {
+    const auto input_lqp = PredicateNode::make(hundred_element_in_expression, node);
+    const auto result_lqp = StrategyBaseTest::apply_rule(rule, input_lqp);
+    EXPECT_EQ(result_lqp, input_lqp);
+  }
+}
+
+TEST_F(InExpressionRewriteRuleTest, DisjunctionStrategy) {
+  auto rule = std::make_shared<InExpressionRewriteRule>();
+  rule->strategy = InExpressionRewriteRule::Strategy::Disjunction;
+
+  {
+    const auto input_lqp = PredicateNode::make(single_element_in_expression, node);
+    const auto result_lqp = StrategyBaseTest::apply_rule(rule, input_lqp);
+    const auto expected_lqp = PredicateNode::make(equals_(col_a, 1), node);
+    EXPECT_LQP_EQ(result_lqp, expected_lqp);
+  }
+
+  {
+    const auto input_lqp = PredicateNode::make(five_element_in_expression, node);
+    const auto result_lqp = StrategyBaseTest::apply_rule(rule, input_lqp);
+
+    // clang-format off
+    // Account for different order produced by ExpressionUnorderedSet
```

Comment:
uargh

Reply:
Yepp, but I did not want to add a superfluous sorting step just for making the test nicer. Any idea how to handle this?

Reply:
No, that is probably the best we can do.